wandb/sdk/launch/agent/agent.py

```diff
@@ -546,7 +551,17 @@ def _check_run_finished(self, job_tracker: JobAndRunStatus) -> bool:
         known_error = False
         try:
             run = job_tracker.run
-            status = run.get_status().state
+            try:
+                status = run.get_status().state
+            except ApiException:
```

Comment:
This check and requeue logic will change based on the SubmittedRun type (k8s, Sagemaker etc). I think it would be cleaner to implement the logic for catching an exception in the Kubernetes submitted run `get_status` function. The get_status function could then return a state indicating that the pod/instance has gone down.

Reply:
Also, the kubernetes runner would also raise this exception if the whole job vanished, so we'll want to consider that as well.